src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -25,6 +30,27 @@
   @NonNull
   private IexCloudClient iexCloudClient;
 
+  /**
+   * Valid range values documentation: https://iexcloud.io/docs/api/#historical-prices.
+   */
+  private final List<String> validRangeValues = Arrays.asList("max", "5y", "2y", "1y", "ytd", "6m",
+      "3m", "1m", "1mm", "5d", "5dm", "date", "dynamic");
+
+  private boolean rangeIsValid(final String range) {
+    return validRangeValues.contains(range.toLowerCase());
+  }
+
+  private final DateTimeFormatter dateFormatter = DateTimeFormatter.BASIC_ISO_DATE;
+
+  private boolean dateIsValid(final String date) {
+    try {
```

Comment:
Generally speaking, using a try/catch to drive logic isn't desirable. Instead, try to find some sort of method of "is parsable", instead of trying to use a try/catch

Reply:
I'm having a hard time finding an effective way to validate the date without a try/catch block. Virtually all approaches I've found seem to utilize a try/catch block. I think I could add Apache Commons Validator as a dependency to avoid the try/catch, but I'm not sure if that's more desirable.
References:
https://www.baeldung.com/java-string-valid-date
https://docs.oracle.com/javase/tutorial/datetime/iso/format.html